### PR TITLE
Add cascading config list

### PR DESCRIPTION
**Pull request by wintron04**
- [x] Cascading config list
- [x] gnupg dir check
- [x] KVM virt checker
- [x] Use ENV var for boxed plugin conf
- [x] Color for Log messages

### CODE REVIEW

File: src/berry_mill/mill.py [hunk 1]
```diff
@@ -110,16 +163,23 @@ def run(self) -> None:
...
             # parameter "cross" implies a amd64 host and an arm64 target-arch
             if self.args.cross:
-                self.args.arch = "arm64"              
+                self.args.arch = "arm64"
+
+            if not self.args.local and not self.args.ignore_nested:
+                if not has_virtualization():
+                    log.info("Berrymill currently cannot detect wether you run it in a virtual environment or on a bare metal")
+                    log.warning(no_nested_warning)
+                    return
```
Comment (isbm) [inline src/berry_mill/mill.py:175]: Why two `if` clauses if it is just one?

File: src/berry_mill/mill.py [hunk 2]
```diff
@@ -78,6 +98,24 @@ def __init__(self):
...
+            raise Exception("Appliance Path not found")
+
+        self._tmp_appliance_dir: str = mkdtemp(prefix="berrymill-tmp", dir="/tmp")
+        self._appliance_abs_path: str = os.path.join(os.getcwd(), self._appliance_descr)
+        self._appliance_tmp_pth: str = os.path.join(self._tmp_appliance_dir, self._appliance_descr)
+
+        try:
+            self._construct_final_appliance()
+        except Exception as e:
+            self.cleanup()
```
Comment (isbm) [inline src/berry_mill/mill.py:117]: This belongs to `finally` block.

**Review by isbm — APPROVED** [review]
@wintron04 could you please look for a few small changes?

**Review by isbm — CHANGES_REQUESTED** [review]
I chose "approve" by a mistake, sorry.

File: src/berry_mill/mill.py [hunk 3]
```diff
@@ -78,6 +98,23 @@ def __init__(self):
...
                     self._appliance_descr = pth
                     break
+        
+        if not self._appliance_descr:
+            raise Exception("Appliance description was not found.")
+        
+        if self._appliance_path:
+            os.chdir(self._appliance_path)
+        else:
+            raise Exception("Appliance Path not found")
```
Comment (isbm) [inline src/berry_mill/mill.py:108]: I would raise or chdir instead:
```python
if not self._appliance_path:
    raise Exception("AAAA!")
os.chdir(self._appliance_path)
```

Less complex to comprehend and saves one code line.

File: src/berry_mill/mill.py [hunk 4]
```diff
@@ -99,6 +136,19 @@ def _init_local_repos(self) -> None:
...
+        """
+        Constructs final kiwi appliance which is used by kiwi
+        1. moves the appliance description to a tmp dir, so kiwi wont use this "wrong" one
+        2. Constructs the right appliance and safes it named as the one passed to berrymill orginially
+        """
+        final_rendered_xml_string = Loader().load(self._appliance_abspath)
+        
+        shutil.move(self._appliance_abspath, self._bac_appliance_abspth)
+
+        with open(self._appliance_abspath, "w") as ma:
```
Comment (isbm) [inline src/berry_mill/mill.py:149]: Maybe remove these empty lines?
We need MyPy and autoformatter, after all. 😉

File: test/test_cfgh.py [hunk 5]
```diff
@@ -79,15 +80,15 @@ def test_add_config_that_exists(self, config_handler: ConfigHandler):
         """ Check that file path was added """
         assert temp_file_path in config_handler._cfg
 
-    def test_add_config_that_not_exists(self, capsys: CaptureFixture, config_handler: ConfigHandler):
+    def test_add_config_that_not_exists(self, caplog: LogCaptureFixture, config_handler: ConfigHandler):
         """ Set non existent path """
         non_existent_config_path: str = "/path/to/non/existent/config.cfg"
         """ add non existent path """
```
Comment (isbm) [inline test/test_cfgh.py:86]: BTW, why you use docstrings as commends? This is not really good. It should be:

```python
def test_something():
    """
    This test tests something
    """
    # Set things
    a:str = "bs"
    
    # Add something
    b:str = "bs bs"
```

...etc. Could you please also clean that up in this PR still? (everywhere).

**Review by isbm — CHANGES_REQUESTED** [review]
@wintron04 some more 😉

File: test/test_cfgh.py (hunk 5 above)
Comment (wintron04) [reply test/test_cfgh.py:86]: I have no idea why they're there 
I just refactored the tests without really looking at them

But I can cleanup yes

File: src/berry_mill/mill.py (hunk 4 above)
Comment (wintron04) [reply src/berry_mill/mill.py:149]: That is a PR for another time

Comment (wintron04) [reply src/berry_mill/mill.py:149]: Then I can do everything at once

File: src/berry_mill/mill.py [hunk 6]
```diff
@@ -1,17 +1,34 @@
...
+
+no_nested_warning: str = str(
+"""
+Nested virtualization is NOT enabled. This can cause the build to fail
+as a virtual enviroment using qemu is utilized to build the image.
+
+You can either: 
+
+Enable nested virtualization, build locally, or, use --ignore-nested when you are
+sure that you are not running berrymill inside a virtual machine
```
[the remark this answers is not in the document]
Comment (isbm) [reply src/berry_mill/mill.py:29]: Issue is available here: https://github.com/isbm/berrymill/issues/48

[the remark this answers is not in the document]
Comment (wintron04) [reply src/berry_mill/mill.py:29]: I will look at that

**Pull request merged**